datafusion-postgres/src/handlers.rs

```diff
@@ -468,6 +488,17 @@ impl SimpleQueryHandler for DfSessionService {
                 self.check_query_permission(client, &query).await?;
             }
 
+            // Call query hooks with the parsed statement
+            for hook in &self.query_hooks {
+                let wrapped_statement = Statement::Statement(Box::new(statement.clone()));
```

Comment:
Do we still need wrap here? Can we feed `statement` directly to `hook.handle_query`?

Reply:
I was simply felt we would want to expose the datafusion Statement and types to users instead of the sqlparser's. But not wrapping might also be cleaner. I've updated the code.